### PR TITLE
Update install instructions with demo values

### PR DESCRIPTION
**Pull request by mandarjog**
1. Update helm instructions to use appropriate values files.
2. Make a note that this configuration is not meant performance testing or production use.

### CODE REVIEW

File: content/docs/setup/kubernetes/helm-install/index.md [hunk 1]
```diff
@@ -101,11 +101,15 @@ Choose this option if your cluster doesn't have [Tiller](https://github.com/kube
     56
     {{< /text >}}
 
-1. Render and apply Istio's core components:
+1. Render and apply Istio's core components in demo mode with mTLS enabled:
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/helm-install/index.md]: mTLS -> mutual TLS

File: content/docs/setup/kubernetes/helm-install/index.md [hunk 2]
```diff
@@ -166,11 +170,15 @@ to manage the lifecycle of Istio.
     56
     {{< /text >}}
 
-1. Install the `istio` chart:
+1. Install the `istio` chart in demo mode with mTLS enabled:
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/helm-install/index.md]: mTLS -> mutual TLS

File: content/docs/setup/kubernetes/helm-install/index.md [hunk 3]
```diff
@@ -166,11 +170,15 @@ to manage the lifecycle of Istio.
...
+
+{{< tip >}}
+This mode is designed to showcase Istio functionality with modest resource requirements. It enables high levels of tracing and access logging so it is not suitable for performance tests. You may use [values.yaml from the istio/tools repo](https://github.com/istio/tools/tree/master/perf/istio) to performance test Istio.
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/helm-install/index.md]: Here and in the other place, change:

You may use this [streamlined `values.yaml`](https://github.com/istio/tools/tree/master/per/istio) file to test Istio's performance.

**Comment by linsun** [comment]
@frankbu has recently reworked this page.  I think this is no longer needed.

File: content/docs/setup/kubernetes/helm-install/index.md [hunk 4]
```diff
@@ -101,11 +101,15 @@ Choose this option if your cluster doesn't have [Tiller](https://github.com/kube
...
+1. Render and apply Istio's core components in demo mode with mTLS enabled:
+
+{{< tip >}}
+This mode is designed to showcase Istio functionality with modest resource requirements. It enables high levels of tracing and access logging so it is not suitable for performance tests. You may use [values.yaml from the istio/tools repo](https://github.com/istio/tools/tree/master/perf/istio) to performance test Istio.
+{{< tip >}}
 
     {{< text bash >}}
 
-    $ helm template $HOME/istio-fetch/istio --name istio --namespace istio-system | kubectl apply -f -
+    $ helm template $HOME/istio-fetch/istio --name istio --namespace istio-system  --values $HOME/istio-fetch/istio/values-istio-demo-common.yaml --values $HOME/istio-fetch/istio/values-istio-demo-auth.yaml | kubectl apply -f -
```
Comment (linsun) [inline content/docs/setup/kubernetes/helm-install/index.md]: this is no longer needed now :)

**Comment by linsun** [comment]
assign @frankbu to help triage this. Thanks!

**Comment by frankbu** [comment]
This is now being addressed by formally describing the various built-in profiles.

I'll add this part to the **demo** profile description: `This mode is designed to showcase Istio functionality with modest resource requirements. It enables high levels of tracing and access logging so it is not suitable for performance tests.`, but otherwise this PR is no longer applicable.

**Pull request closed without merging**